py-polars/polars/series/series.py

```diff
@@ -1084,13 +1084,7 @@ def __truediv__(self, other: Any) -> Series | Expr:
             msg = "first cast to integer before dividing datelike dtypes"
             raise TypeError(msg)
 
-        # this branch is exactly the floordiv function without rounding the floats
-        if self.dtype.is_float() or self.dtype == Decimal:
-            as_float = self
-        else:
-            as_float = self._recursive_cast_to_dtype(Float64())
-
-        return as_float._arithmetic(other, "div", "div_<>")
+        return self._arithmetic(other, "div", "div_<>")
```

Comment:
Series division had a `cast(Float64)` on the Python side - I have moved this to the rust `impl Div for &Series`